test/plugin/test_out_http.rb

```diff
@@ -0,0 +1,351 @@
+# coding: utf-8
+require_relative "../helper"
+require 'fluent/test/driver/output'
+require 'fluent/plugin/out_http'
+
+require 'webrick'
+require 'webrick/https'
+require 'net/http'
+require 'uri'
+require 'json'
+
+# WEBrick's ProcHandler doesn't handle PUT by default
+module WEBrick::HTTPServlet
+  class ProcHandler < AbstractServlet
+    alias do_PUT  do_GET
+  end
+end
+
+class HTTPOutputTest < Test::Unit::TestCase
+  include Fluent::Test::Helpers
+
+  TMP_DIR = File.join(__dir__, "../tmp/out_http#{ENV['TEST_ENV_NUMBER']}")
+  DEFAULT_LOGGER = ::WEBrick::Log.new(::STDOUT, ::WEBrick::BasicLog::FATAL)
+
+  class << self
+    # Use class variable to reduce server start/shutdown time
+    def startup
+      @@result = nil
+      @@auth_handler = nil
+      @@http_server_thread = nil
+    end
+
+    def shutdown
+      @@http_server_thread.kill
+      @@http_server_thread.join
+    rescue
+    end
+  end
+
+  def server_port
+    19880
```

Comment:
unused_port is better?

Reply:
I'm not sure which is the better. This is fixed value, not dynamic value like helper's unused_port.
So this naming is also ok for me.